File: 基礎題庫/Contents/c636/c636.py
import sys

symbol = ["豬", "鼠", "牛", "虎", "兔", "龍", "蛇", "馬", "羊", "猴", "雞", "狗"]

for n in sys.stdin:
    n = int(n)

    if n < 0:
        n += 121
    
    print(symbol[n % 12])

File: 基礎題庫/Contents/c636/c636.cpp
#include<iostream>

using namespace std;

char symbol[12][5] = {"豬", "鼠", "牛", "虎", "兔", "龍", "蛇", "馬", "羊", "猴", "雞", "狗"};

int main(void)
{
	int n;
	
	while(cin >> n)
		cout << symbol[(n < 0 ? n + 121 : n) % 12] << endl;
	
	return 0;
}
